File: business-management-software-frontend/src/services/CategoryService.js
import axios from "axios";

const category_API_BASE_URL = "http://localhost:8080/api";

class CategoryService {
  createCategory(category) {
    return axios.post(category_API_BASE_URL + "/categories", category);
  }
  getCategories() {
    return axios.get(category_API_BASE_URL + "/categories");
  }
  getCategoryById(categoryId) {
    return axios.get(category_API_BASE_URL + "/category/" + categoryId);
  }
  updateCategory(category, categoryId) {
    return axios.put(
      category_API_BASE_URL + "/category/" + categoryId,
      category
    );
  }
  countProductsByCategory(categoryId) {
    const url = `${category_API_BASE_URL}/categories/productsCount/${categoryId}`;
    return axios.get(url);
  }

  searchCategory(searchTerm) {
    return axios.get(
      category_API_BASE_URL + "/categories/search/" + searchTerm
    );
  }

  deleteCategory(categoryId) {
    return axios.delete(category_API_BASE_URL + "/category/" + categoryId);
  }
}
export default new CategoryService();

File: BusinessManagementSoftwareBackend/src/main/java/com/ali/BusinessManagementSoftwareBackend/services/categories/CategoryService.java
package com.ali.BusinessManagementSoftwareBackend.services.categories;

import com.ali.BusinessManagementSoftwareBackend.dto.CategoryDto;

import java.io.IOException;
import java.util.List;

public interface CategoryService {
    boolean createCategory(CategoryDto categoryDto) throws IOException;
    List<CategoryDto> getCategories();
    void deactivateCategory(Long id);
    CategoryDto getCategoryById(Long id);
    boolean updateCategory(Long id, CategoryDto categoryDto);
    public Long countProductsByCategory(Long categoryId) ;
    public List<CategoryDto> searchCategory(String name) ;



    }
